app/services/ingest_vocab_service.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+require 'csv'
+
+class IngestVocabService
+  delegate :metadata_adapter, to: :change_set_persister
+  delegate :query_service, to: :metadata_adapter
+  attr_reader :file, :name, :columns, :change_set_persister, :logger
+  def initialize(change_set_persister, file, name, columns, logger = Logger.new(STDOUT))
+    @file = file
+    @name = name
+    @columns = columns
+    @change_set_persister = change_set_persister
+    @logger = logger
+  end
+
+  def vocab_queries
+    @vocab_queries ||= QueryAdapter.new(query_service: query_service, model: EphemeraVocabulary)
+  end
+
+  def vocab_persistence
+    @vocab_persistence ||= PersistenceAdapter.new(change_set_persister: change_set_persister, model: EphemeraVocabulary)
+  end
+
+  def term_queries
+    @term_queries ||= QueryAdapter.new(query_service: query_service, model: EphemeraTerm)
+  end
+
+  def term_persistence
+    @term_persistence ||= PersistenceAdapter.new(change_set_persister: change_set_persister, model: EphemeraTerm)
+  end
+
+  def ingest_vocabulary(label:, parent_vocab: nil)
+    return unless label
+    category = vocab_queries.find_with(FindEphemeraVocabularyByLabel, label: label).first
```

Comment:
Soon this'll be able to become `vocab_queries.custom_queries.find_ephemera_vocabulary_by_label(label: label)`, so I'm not 100% sure the `QueryAdapter` abstraction is necessary.